ui/src/importer/importerTaskDetails/ImportedDocuments.js

```diff
@@ -53,23 +92,132 @@ export class ImportedDocuments extends React.Component {
     if (!importCompleted) {
       const nextEntry = _get(data, 'loaded_entries', 0);
       const response = await importerApi.check(taskId, nextEntry);
+      const responseData = response.data;
+
+      if (responseData) {
+        responseData.records = _get(responseData, 'records', []);
+      }
       if (response.data.status !== 'RUNNING') {
         this.setState({
           importCompleted: true,
           isLoading: false,
-          data: response.data,
+          data: responseData,
         });
       } else {
         this.setState({
-          data: response.data,
           isLoading: true,
+          data: responseData,
         });
       }
+      this.modeLabel();
+      this.statusLabel();
+      this.calculateStatistics();
     } else {
       this.intervalId && clearInterval(this.intervalId);
     }
   };
 
+  modeLabel = () => {
+    let { data, mode } = this.state;
+    switch (data.mode) {
+      case 'IMPORT':
+        mode = (
+          <Label color="blue" basic>
+            Import
+          </Label>
+        );
+        break;
+      case 'DELETE':
+        mode = (
+          <Label color="red" basic>
+            Delete
+          </Label>
+        );
+        break;
+      case 'PREVIEW_IMPORT':
+        mode = (
+          <Label color="teal" basic>
+            Preview (import)
+          </Label>
+        );
+        break;
+      case 'PREVIEW_DELETE':
+        mode = (
+          <Label color="teal" basic>
+            Preview (delete)
+          </Label>
+        );
+        break;
+      default:
+        mode = (
+          <Label color="black" basic>
+            Invalid mode
+          </Label>
+        );
+        break;
+    }
+    this.setState({
+      mode: mode,
+    });
+  };
+
+  statusLabel = () => {
+    let { data, status, isLoading } = this.state;
+    !data
+      ? (status = <Label color="grey">Fetching status</Label>)
```

Comment:
I am still trying to understand if we can improve code readability here. My feeling is that, since we check the `data` and the `data.status`, I wouldn't mix the things.
```
if (!data) { // or !!data?
  status = <Label color="grey">Fetching status</Label>
} else {
  switch...
    RUNNING
    SUCCEEDED
    FAILED
    CANCELLED
}
```
Since the content of these cmps is static, it doesn't depend on the status, I would avoid to re-create each time.
You could create some consts on the top of the cmp:
```
STATUS_CMPS = {
  'RUNNING' = <Label...,
  '...'
}
```
and then use it in the render method:
```
cmp = STATUS_CMPS[data.status];
```

Same for the mode above. You would avoid also to have a *fat* state by putting much less cmps inside.

Reply:
another alternative to if / else is using guard clauses so just return it early if a a certain condition is met so that you do not need to wrap it in an else statement 

```
if(!data) {
return <Label .....>
}

switch () {
......
}

```